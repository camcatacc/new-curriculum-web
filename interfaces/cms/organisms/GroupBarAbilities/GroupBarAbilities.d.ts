import type { ContentfulEntry } from "interfaces/cms/contentful";
import type { CmsBarAbility } from "interfaces/cms/molecules/BarAbility";

export interface CmsGroupBarAbilities {
	title: string;
	abilities: ContentfulEntry<CmsBarAbility>[];
	titleColor: string;
	barColor: string;
	fontColor?: string;
}
