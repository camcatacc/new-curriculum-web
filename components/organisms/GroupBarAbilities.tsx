// Modules
import React from "react";

// Elements
import BarAbility from "components/molecules/BarAbility";
import { Typography } from "@material-ui/core";

// Definitions
import type { BarAbilityProps, FormatBarAbilityProps, UnformattedBarAbilityProps } from "components/molecules/BarAbility";

export interface GroupBarAbilitiesProps {
	abilities: BarAbilityProps[];
	title?: string;
}

export interface GroupBarUnformattedAbilitiesProps extends FormatBarAbilityProps {
	abilities: UnformattedBarAbilityProps[];
	title?: string;
}

// Element
const GroupBarAbilities = ({ abilities, title }: GroupBarAbilitiesProps | GroupBarUnformattedAbilitiesProps) => {
	return (
		<div className="flex flex-col flex-1 gap-4">
			{title ? <Typography variant="h3">{title}</Typography> : null}
			{abilities.map((hab, ind) => (
				<div className="flex-1" key={`hab${ind}`}>
					<BarAbility {...hab} titleStyle={{ backgroundColor: "red", fontColor: "blue" }} title={"bu"} barStyle={{ backgroundColor: "yellow" }} />
				</div>
			))}
		</div>
	);
};

export default GroupBarAbilities;
