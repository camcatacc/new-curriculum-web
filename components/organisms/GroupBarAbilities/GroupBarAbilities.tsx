// Modules
import React from "react";

// Elements
import BarAbility from "components/molecules/BarAbility/BarAbility";
import { Typography } from "@material-ui/core";

// Definitions
import type { CmsBarAbility } from "interfaces/cms/molecules/BarAbility";

export interface GroupBarAbilitiesProps {
	title: string;
	abilities: CmsBarAbility[];
	titleColor: string;
	barColor: string;
	fontColor?: string;
}

// Element
const GroupBarAbilities = ({ abilities, title, titleColor, barColor, fontColor }: GroupBarAbilitiesProps) => (
	<div className="flex flex-col flex-1 gap-4">
		<Typography variant="h3" align="center">
			{title}
		</Typography>
		{abilities.map((ab, ind) => (
			<BarAbility {...ab} key={`ab${ind}`} barColor={ab.barColor ?? barColor} fontColor={ab.fontColor ?? fontColor} titleColor={ab.titleColor ?? titleColor} />
		))}
	</div>
);

export default GroupBarAbilities;
