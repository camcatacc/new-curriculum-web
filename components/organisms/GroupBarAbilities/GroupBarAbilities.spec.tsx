// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import GroupBarAbilities from "components/organisms/GroupBarAbilities/GroupBarAbilities";

// Definitions
import type { GroupBarAbilitiesProps } from "components/organisms/GroupBarAbilities/GroupBarAbilities";

// Default Props
const defaultProps: GroupBarAbilitiesProps = {
	abilities: [
		{ percentage: 80, name: "Cool" },
		{ percentage: 80, name: "Cool 2" }
	],
	barColor: "white",
	title: "Coolest abilities",
	titleColor: "red"
};

// Tests
describe("GroupBarAbilities", () => {
	it("Renders as expected", () => {
		const { container } = render(<GroupBarAbilities {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
