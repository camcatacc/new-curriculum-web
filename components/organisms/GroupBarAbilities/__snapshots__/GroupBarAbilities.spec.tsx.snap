// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupBarAbilities Renders as expected 1`] = `
<div>
  <div
    class="flex-1"
  >
    <div
      class="mb-5"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
      >
        Coolest abilities
      </h3>
    </div>
    <div
      class="flex flex-col gap-3"
    >
      <div
        class="flex flex-row justify-center"
      >
        <div
          class="w-40 md:w-32 truncate"
        >
          <div
            class="sc-dkPtRN kkgdvy pl-2 pr-2 pt-1 pb-1"
          >
            <p
              class="MuiTypography-root sc-hKwDye dsVJSv w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
              font-size="1rem"
              fontcolor="white"
            >
              Cool
            </p>
          </div>
        </div>
        <div
          class="w-full md:w-80 "
        >
          <div
            class="bg-gray-300 h-full"
          >
            <div
              class="sc-bdvvtL iKfuUE pr-4 h-full flex flex-col justify-center"
            >
              <div>
                <p
                  class="MuiTypography-root sc-gsDKAQ etFjmY MuiTypography-body1 MuiTypography-alignRight"
                  font-size="1rem"
                  fontcolor="white"
                >
                  80%
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-row justify-center"
      >
        <div
          class="w-40 md:w-32 truncate"
        >
          <div
            class="sc-dkPtRN kkgdvy pl-2 pr-2 pt-1 pb-1"
          >
            <p
              class="MuiTypography-root sc-hKwDye dsVJSv w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
              font-size="1rem"
              fontcolor="white"
            >
              Cool 2
            </p>
          </div>
        </div>
        <div
          class="w-full md:w-80 "
        >
          <div
            class="bg-gray-300 h-full"
          >
            <div
              class="sc-bdvvtL iKfuUE pr-4 h-full flex flex-col justify-center"
            >
              <div>
                <p
                  class="MuiTypography-root sc-gsDKAQ etFjmY MuiTypography-body1 MuiTypography-alignRight"
                  font-size="1rem"
                  fontcolor="white"
                >
                  80%
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
